protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/generate/SourceCodeGenerator.java

```diff
@@ -104,6 +105,11 @@ public boolean generate(GenerationRequest request) throws ResolutionException, I
       }
     }
 
+    //if (resolvedPlugins.isEmpty() && hasPlugins(request) && pluginsOptional(request)) {
```

Comment:
Planning to replace with checks that all langs are disabled and no plugins are provided. This will default to being an error but can be ignored via a parameter I plan to add.